### PR TITLE
✨ Add mcdoc validation for NBT

### LINKED ISSUE #1210: Add mcdoc validation for NBT
~~blocked by #1187~~

### PR DESCRIPTION
**Pull request by TheAfroOfDoom**
Closes #1210

### CODE REVIEW

File: packages/mcdoc/src/runtime/checker/index.ts [hunk 1]
```diff
@@ -1188,12 +1193,24 @@ function getValueType(
 	}
 }
 
-export function getDefaultErrorReporter<T>(
+function getErrorRange<T extends AstNode>(
+	node: RuntimeNode<T>,
+	error: McdocCheckerError<T>['kind'],
+): Range {
```
Comment (misode) [inline packages/mcdoc/src/runtime/checker/index.ts:1199]: I was confused why the build was failing, but this explains it. You're assuming here that `T` will always be an `AstNode`. The runtime checker does not assume that (as is demonstrated by the runtime checker tests, which use simple JS objects). Therefore, a `reportError` options will still need to be required, but you can just simplify those to not check the node type. Reverting this latest commit is probably the easiest.

**Review by misode — CHANGES_REQUESTED** [review]

File: packages/mcdoc/src/runtime/checker/index.ts (hunk 1 above)
Comment (TheAfroOfDoom) [reply packages/mcdoc/src/runtime/checker/index.ts:1199]: this commit is incomplete, I only pushed it to be able to continue working on it on my laptop 😄

**Review by misode — APPROVED** [review]
Looks complete to me. I made a few fixes while I was testing it.

File: packages/mcdoc/src/runtime/checker/index.ts [hunk 2]
```diff
@@ -1469,9 +1470,24 @@ function getValueType(
 	}
 }
 
+export function getErrorRangeDefault<T extends AstNode>(
```
Comment (NeunEinser) [inline packages/mcdoc/src/runtime/checker/index.ts:1473]: should be `getDefaultErrorRange` (see also `getDefaultErrorReporter`)

**Review by NeunEinser — COMMENTED** [review]
Nice work, looks good!

**Pull request merged**